testing/README.md

```diff
@@ -0,0 +1,355 @@
+# `import {Testing} from 'Shopify'`
+
+This guide looks at our recommended tools and conventions for testing JavaScript at Shopify. The primary focus of this guide is unit testing, but you can find some discussion of other types of testing in the [types of testing section](#types-of-testing).
+
+## Table of contents
+
+1. [Tools](#tools)
+1. [Style](#style)
+1. [Best practices](#best-practices)
+1. [Types of testing](#types-of-testing)
+1. [React](#react)
+1. [Resources](#resources)
+
+
+
+## Tools
+
+We recommend using the following tools in your tests:
+
+- [**mocha**](https://mochajs.org/) is our preferred test framework. It provides the basic scaffolding for tests including hooks, excellent support for asynchronous code, and more.
+
+- [**chai**](http://chaijs.com/) is our preferred assertion library. It provides many expressive assertions that have descriptive error messages in the case of test failures.
+
+- [**sinon**](http://sinonjs.org/) is our preferred library for handling spies and mocks in JavaScript.
+
+These tools are commonly used together and have many excellent online resources, including solid documentation. In addition to the above, you can use [chai plugins](http://chaijs.com/plugins/) to provide more meaningful assertions. Use caution, however: only include a plugin where it materially helps improve the legibility of tests (otherwise, the burden of other developers tracking down a particular assertion outweighs the benefits you get from having it). [chai-as-promised](http://chaijs.com/plugins/chai-as-promised/) is an excellent example of a chai plugin that makes otherwise awkward tests much easier to read.
+
+In addition to the testing framework, you generally need a test *runner*. Which runner to use will depend on the nature of your project:
+
+- For pure-JavaScript projects that do not need access to the DOM, avoid using a test runner and simply run the `mocha` CLI instead (if you need a simple DOM without actual rendering, you can also use [cheerio](https://github.com/cheeriojs/cheerio) to construct a fake DOM).
+
+- For Non-Rails projects that require a full DOM, use [Karma](https://karma-runner.github.io/1.0/index.html). It integrates well with other tools you are likely using, such as Webpack and code coverage tools.
```

Comment:
I'm not we should explicitly mention Webpack since there seems to be a competition between bundlers heating up. I know ppl who are absolutely opposed to things that Webpack does such as overloading `require`. Might be better just to say "bundlers and code coverage tools".